sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -94,6 +94,11 @@ class ZeCall {
 };
 std::mutex ZeCall::GlobalLock;
 
+// The global mutex to get rid of calling zeEventCreate from simultaneous
+// threads in multithreading application (prohibited by Level Zero spec).
+// Without the lock, RT finishes with CL_INVALID_VALUE from piEventCreate
+static std::mutex EventCreateMutex;
```

Comment:
https://spec.oneapi.io/level-zero/latest/core/api.html?highlight=zeeventcreate#_CPPv413zeEventCreate22ze_event_pool_handle_tPK15ze_event_desc_tP17ze_event_handle_t:

> The application must not call this function from simultaneous threads with the same event pool handle.

We shouldn't need a global lock, just lock the pool. Please sync with @againull who is adding locks to all PI objects in https://github.com/intel/llvm/pull/4586